bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java

```diff
@@ -0,0 +1,193 @@
+        channelrefreshActions.forEach((channelUID, action) -> action.accept(thermostat));
+    }
+
+    private void updateManualSetpoint(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_MANUALSETPOINT,
+                new QuantityType<Temperature>(thermostat.manualModeSetpoint / (double) 100, SIUnits.CELSIUS));
+    }
+
+    private void updateBoostEndTime(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_BOOSTENDTIME,
+                new DateTimeType(ZonedDateTime.ofInstant(thermostat.boostEndTime.toInstant(), ZoneId.systemDefault())));
+    }
+
+    private void updateComfortEndTime(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_COMFORTENDTIME, new DateTimeType(
+                ZonedDateTime.ofInstant(thermostat.comfortEndTime.toInstant(), ZoneId.systemDefault())));
+    }
+
+    private void updateComfortSetpoint(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_COMFORTSETPOINT,
+                new QuantityType<Temperature>(thermostat.comfortSetpoint / (double) 100, SIUnits.CELSIUS));
+    }
+
+    private void updateRegulationMode(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_REGULATIONMODE,
+                StringType.valueOf(getRegulationMode(thermostat.regulationMode)));
+    }
+
+    private void updateThermostatName(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_THERMOSTATNAME, StringType.valueOf(thermostat.thermostatName));
+    }
+
+    private void updateFloorTemperature(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_FLOORTEMPERATURE,
+                new QuantityType<Temperature>(thermostat.floorTemperature / (double) 100, SIUnits.CELSIUS));
+    }
+
+    private void updateRoomTemperature(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_ROOMTEMPERATURE,
+                new QuantityType<Temperature>(thermostat.roomTemperature / (double) 100, SIUnits.CELSIUS));
+    }
+
+    private void updateHeating(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_HEATING,
+                thermostat.heating ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+    }
+
+    private void updateOnline(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_ONLINE,
+                thermostat.online ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+    }
+
+    private void updateGroupId(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_GROUPID, new DecimalType(thermostat.groupId));
+    }
+
+    private void updateGroupName(Thermostat thermostat) {
+        updateState(BindingConstants.CHANNEL_OWD5_GROUPNAME, StringType.valueOf(thermostat.groupName));
+    }
+
+    private String getRegulationMode(int regulationMode) {
+        return REGULATION_MODES.get(regulationMode);
+    }
+
+    private static HashMap<Integer, String> createRegulationMap() {
```

Comment:
Declare vars with their interface for the `Collection` types, so this should rather be: 
```suggestion
    private static Map<Integer, String> createRegulationMap() {
```